src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java

```diff
@@ -0,0 +1,122 @@
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCourseEditPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        course = testData.courses.get("InsCrsEdit.CS2104");
+        instructors[0] = testData.instructors.get("InsCrsEdit.helper");
+        instructors[1] = testData.instructors.get("InsCrsEdit.manager");
+        instructors[2] = testData.instructors.get("InsCrsEdit.observer");
+        instructors[3] = testData.instructors.get("InsCrsEdit.coowner");
+        instructors[4] = testData.instructors.get("InsCrsEdit.tutor");
+    }
+
+    @Test
+    public void testAll() {
+        ______TS("verify cannot edit without privilege");
+        // log in as instructor with no edit privilege
+        AppUrl url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_EDIT_PAGE)
+                .withUserId(instructors[2].googleId)
+                .withCourseId(course.getId());
+        loginAdminToPage(url, InstructorHomePage.class);
+        InstructorCourseEditPage editPage = AppPage.getNewPageInstance(browser, url, InstructorCourseEditPage.class);
+
+        editPage.verifyCourseNotEditable();
+        editPage.verifyInstructorsNotEditable();
+        editPage.verifyAddInstructorNotAllowed();
+
+        ______TS("verify loaded data");
+        // re-log in as instructor with edit privilege
+        url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_EDIT_PAGE)
+                .withUserId(instructors[3].googleId)
+                .withCourseId(course.getId());
+        editPage = AppPage.getNewPageInstance(browser, url, InstructorCourseEditPage.class);
+
+        editPage.verifyCourseDetails(course);
+
+        editPage.verifyInstructorDetails(1, instructors[0]);
+        editPage.verifyInstructorDetails(2, instructors[1]);
+        editPage.verifyInstructorDetails(3, instructors[2]);
+        editPage.verifyInstructorDetails(4, instructors[3]);
+        editPage.verifyInstructorDetails(5, instructors[4]);
+
+        ______TS("add instructor");
+        InstructorAttributes newInstructor = InstructorAttributes
+                .builder(course.getId(), "InsCrsEdit.test@gmail.tmt")
+                .withName("Teammates Test")
+                .withIsDisplayedToStudents(true)
+                .withDisplayedName("Instructor")
+                .withRole("Tutor")
+                .build();
+
+        editPage.addInstructor(newInstructor);
+        editPage.verifyStatusMessage("\"The instructor Teammates Test has been added successfully. "
+                + "An email containing how to 'join' this course will be sent to InsCrsEdit.test@gmail.tmt"
+                + " in a few minutes.\"");
+        editPage.verifyInstructorDetails(6, newInstructor);
+
+        ______TS("resend invite");
+        editPage.resendInstructorInvite(6);
+        editPage.verifyStatusMessage("An email has been sent to InsCrsEdit.test@gmail.tmt");
+
+        ______TS("edit instructor");
```

Comment:
Same for this test case. The backend is not verified. Actually this point holds throughout for the other test cases!